src/Cli/dotnet/CommonOptions.cs

```diff
@@ -69,14 +67,14 @@ public static IEnumerable<string> RuntimeArgFunc(string rid)
         new DynamicForwardedOption<string>("--runtime", "-r")
         {
             HelpName = RuntimeArgName
-        }.ForwardAsMany(RuntimeArgFunc)
+        }.ForwardAsMany(RuntimeArgFunc!)
         .AddCompletions(CliCompletion.RunTimesFromProjectFile);
 
     public static Option<string> LongFormRuntimeOption =
         new DynamicForwardedOption<string>("--runtime")
         {
             HelpName = RuntimeArgName
-        }.ForwardAsMany(RuntimeArgFunc)
+        }.ForwardAsMany(RuntimeArgFunc!)
```

Comment:
why did you need the `!` for these? The static function is known, isn't it?

Reply:
```csharp
public static Option<T> ForwardAsMany<T>(this ForwardedOption<T> option, Func<T?, IEnumerable<string>> format);
```

This requires that the delegate have a `string?` parameter, presumably more fallout of `ParseResult` design. The `RuntimeArgFunc` though depends on having a non-null parameter. The parameter is immediately passed to `GetArchFromRid` and that throws on `null`.